File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpPlaceDiscreteRequest.java
package calebxzhou.rdi.mc.common2.mcp;

import java.util.List;
import java.util.Map;

public record RMcpPlaceDiscreteRequest(
        String blockId,
        List<Target> targets,
        boolean dryRun
) {
    public record Target(
            RMcpBlockPosData pos,
            Map<String, String> state
    ) {
    }
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpPlaceDiscreteRequest.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpPlaceDiscreteRequest(
    val blockId: String,
    val targets: List<Target>,
    val dryRun: Boolean
) {
    @JvmRecord
    data class Target(
        val pos: RBlockPos,
        val state: Map<String, String>?
    )
}
